### PR TITLE
feat(ui): Add ability to change number of lines shown in pod log viewer

### LINKED ISSUE #6199: Number of lines in logs limited to 100
# Summary

The number of lines in pod logs and deployment logs is limited to 100

# Motivation

This number is hardcoded and it's not possible to change

# Proposal

Implement it in variable in install (parameter) and/or change it in UI

# File
https://github.com/argoproj/argo-cd/blob/b37eee1054e42c873699460dd5e2447c2f9fe5a6/ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx

### PR DESCRIPTION
**Pull request by rbreeze**
Fixes #6199

Signed-off-by: Remington Breeze <remington@breeze.software>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx [hunk 1]
```diff
@@ -37,6 +35,32 @@ export const PodsLogsViewer = (props: PodLogsProps & {fullscreen?: boolean}) =>
     const page = props.page;
     const setPage = props.setPage;
     const [viewPodNames, setViewPodNames] = useState(false);
+    const [logLinesField, setLogLinesField] = useState(null);
```
Comment (alexmt) [inline ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx:38]: I think we are planning to replace `DataLoader` later with a hook-based analog. Can we keep using DataLoader until hook based version is available ?

**Review by alexmt — CHANGES_REQUESTED** [review]

File: ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx (hunk 1 above)
Comment (rbreeze) [reply ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx:38]: Good idea. The `useData` hook is available in the `argo-ux` package now, so I can either switch to that, or use `DataLoader` for now

**Comment by xchan** [comment]
Any update on this PR?

**Comment by BenCoughlan15** [comment]
Would be great to get this in place. 

100 lines makes it pretty unusable.

**Comment by Sinhyeok** [comment]
Any update?

**Comment by crenshaw-dev** [comment]
@rbreeze could you resolve conflicts? I'd be happy to test and try to help get this into 2.5.

**Comment by rbreeze** [comment]
@crenshaw-dev will do!

**Comment by alexec** [comment]
This is superceded.

**Pull request closed without merging**